Tests/Marketplace/marketplace_services.py

```diff
@@ -1579,31 +1614,44 @@ def check_if_exists_in_index(self, index_folder_path):
         finally:
             return task_status, exists_in_index
 
-    def upload_integration_images(self, storage_bucket):
+    def upload_integration_images(self, storage_bucket, diff_files_list):
         """ Uploads pack integrations images to gcs.
 
         The returned result of integration section are defined in issue #19786.
 
         Args:
             storage_bucket (google.cloud.storage.bucket.Bucket): google storage bucket where image will be uploaded.
+            diff_files_list (list): The list of all modified/added files found in the diff
 
         Returns:
             bool: whether the operation succeeded.
             list: list of dictionaries with uploaded pack integration images.
 
         """
         task_status = True
-        uploaded_integration_images = []
+        integration_images = []
+        integration_dirs = []
+        unified_integrations = []
 
         try:
-            pack_local_images = self._search_for_images(target_folder=PackFolders.INTEGRATIONS.value)
-
-            if not pack_local_images:
-                return uploaded_integration_images  # returned empty list if not images found
+            # detect added/modified integration images
+            for file in diff_files_list:
+                if self.is_integration_image(file.a_path):
+                    # the integration dir name will show up in the unified integration file path in content_packs.zip
+                    integration_dirs.append(os.path.basename(os.path.dirname(file.a_path)))
+                elif self.is_unified_integration(file.a_path):
+                    # if the file found in the diff is a unified integration we upload its image
+                    unified_integrations.append(os.path.basename(file.a_path))
```

Comment:
This should be done once (not once for each pack). Maybe a dedicated function to create both lists. This will be called from the upload_pack flow (not from within this function)